src/Sulu/Bundle/AdminBundle/Resources/js/views/List/toolbarActions/UploadToolbarAction.js

```diff
@@ -0,0 +1,307 @@
+        const {accept} = this.options;
+
+        if (accept === undefined || accept === null) {
+            return undefined;
+        }
+
+        if (!Array.isArray(accept)) {
+            throw new Error('The "accept" option must be an array!');
+        }
+
+        return accept;
+    }
+
+    @computed get minSize(): ?number {
+        const {minSize} = this.options;
+
+        if (minSize === undefined || minSize === null) {
+            return undefined;
+        }
+
+        if (typeof minSize !== 'number') {
+            throw new Error('The "minSize" option must be a number!');
+        }
+
+        return minSize;
+    }
+
+    @computed get maxSize(): ?number {
+        const {maxSize} = this.options;
+
+        if (maxSize === undefined || maxSize === null) {
+            return undefined;
+        }
+
+        if (typeof maxSize !== 'number') {
+            throw new Error('The "maxSize" option must be a number!');
+        }
+
+        return maxSize;
+    }
+
+    @computed get multiple(): boolean {
+        const {multiple = false} = this.options;
+
+        if (typeof multiple !== 'boolean') {
+            throw new Error('The "multiple" option must be a boolean!');
+        }
+
+        return multiple;
+    }
+
+    @computed get maxFiles(): ?number {
+        return this.multiple ? undefined : 1;
+    }
+
+    @computed get requestPropertyName(): string {
+        const {requestPropertyName} = this.options;
+
+        if (!requestPropertyName) {
+            return this.multiple ? 'files' : 'file';
+        }
+
+        if (typeof requestPropertyName !== 'string') {
+            throw new Error('The "requestPropertyName" option must be a string!');
+        }
+
+        return requestPropertyName;
+    }
+
+    getToolbarItemConfig() {
+        return {
+            type: 'button',
+            label: this.label,
+            icon: this.icon,
+            onClick: this.handleClick,
+        };
+    }
+
+    getNode() {
+        return (
+            <Dropzone
+                accept={this.accept}
+                key="sulu_admin.upload"
+                maxFiles={this.maxFiles}
+                maxSize={this.maxSize}
+                minSize={this.minSize}
+                multiple={this.multiple}
+                noClick={true}
+                noDrag={true}
+                noKeyboard={true}
+                onDropAccepted={this.handleConfirm}
+                onDropRejected={this.handleError}
+                ref={this.setDropzoneRef}
+            >
+                {({getRootProps, getInputProps}) => {
+                    return (
+                        <div {...getRootProps()}>
+                            <input {...getInputProps()} />
+                        </div>
+                    );
+                }}
+            </Dropzone>
+        );
+    }
+
+    // Method copied from https://stackoverflow.com/q/10420352/7733374
+    getReadableFileSizeString = (fileSizeInBytes: number): string => {
```

Comment:
would be okay for me to put this into the `AdminBundle/Resources/js/utils` folder and use it in the `BytesFieldTransformer` too